src/SCE/sce_engine.c

```diff
@@ -327,6 +328,44 @@ void sce_parameters_allocate_session(struct sce_parameters* v)
 	sce_parameters_set_session(v, sce_session_new());
 }
 
+static void _pipe_try_read_into_string(int fd, struct oscap_string *string, bool *open)
+{
+	// FIXME: Read by larger chunks in the future
+	char readbuf;
+	while (true) {
+		const int read_status = read(fd, &readbuf, 1);
+		if (read_status == 1) {  // successful read
+			if (readbuf == '&') {
+				// & is a special case, we have to "escape" it manually
+				// (all else will eventually get handled by libxml)
+				oscap_string_append_string(string, "&amp;");
+			} else {
+				oscap_string_append_char(string, readbuf);
+			}
+			//printf("Read %i from fd=%i\n", readbuf, fd);
```

Comment:
Can't use debug macros, SCE is outside libopenscap. I can remove the printfs if you want, I figured they'd be useful to whoever was testing/reviewing this.